### PR TITLE
Move build directory of dep to _build

### PR DESCRIPTION
**Pull request by ejguan**
Only tested on my local mac

### CODE REVIEW

File: scripts/_setup-macos.sh [hunk 1]
```diff
@@ -35,7 +35,8 @@ set -x # Print commands that are executed.
 FB_OS_VERSION=v2021.05.10.00
 NPROC=$(sysctl -n hw.physicalcpu)
 COMPILER_FLAGS="-mavx2 -mfma -mavx -mf16c -masm=intel -mlzcnt"
-DEPENDENCY_DIR=${DEPENDENCY_DIR:-$(pwd)}
+DEPENDENCY_DIR=${DEPENDENCY_DIR:-$(pwd)}/_build
```
Comment (wenleix) [inline scripts/_setup-macos.sh:38]: Wondering if it makes sense to do something like 
```
DEPENDENCY_DIR=XXX 
```
 
in `build_mac_dep.sh` ? -- so we can keep this file sync with the Velox repo on ?

Comment (ejguan) [reply scripts/_setup-macos.sh:38]: yeah, doable.

File: scripts/_setup-macos.sh [hunk 2]
```diff
@@ -35,7 +35,6 @@ set -x # Print commands that are executed.
 FB_OS_VERSION=v2021.05.10.00
 NPROC=$(sysctl -n hw.physicalcpu)
 COMPILER_FLAGS="-mavx2 -mfma -mavx -mf16c -masm=intel -mlzcnt"
-DEPENDENCY_DIR=${DEPENDENCY_DIR:-$(pwd)}
```
Comment (wenleix) [inline scripts/_setup-macos.sh:38]: If we define `DEPENDENCY_DIR` before sourcing the `_setup-macos.sh`, do we still need to remove this line? ;)

**Review by wenleix — APPROVED** [review]
Thanks!

File: scripts/_setup-macos.sh (hunk 2 above)
Comment (ejguan) [reply scripts/_setup-macos.sh:38]: Why we need to keep it as `DEPENDENCY_DIR` has been declared.

Comment (wenleix) [reply scripts/_setup-macos.sh:38]: @ejguan In theory to keep the file same as https://github.com/facebookincubator/velox/blob/78159a995038e38576c354348023f50cc49a4864/scripts/setup-macos.sh so in the future it's easy to upgrade if Velox update's the `setup-macos.sh` ?

https://github.com/facebookresearch/torcharrow/blob/3fd191043989d2130d7df93535d795bbd64c3313/scripts/_setup-macos.sh#L16-L18

Comment (ejguan) [reply scripts/_setup-macos.sh:38]: Then, why not delete this file and call the script from velox directly?

Comment (wenleix) [reply scripts/_setup-macos.sh:38]: Guess we can also do this? -- but we don't have Mac CI yet so we don't know whether our Mac build might break when update Velox submodule.

**Pull request merged**